src/main/java/com/trolley/trolley/Invoice.java

```diff
@@ -0,0 +1,311 @@
+    }
+
+    public void setInvoiceNumber(String invoiceNumber) {
+        this.invoiceNumber = invoiceNumber;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public String getStatus() {
+        return status;
+    }
+
+    public void setStatus(String status) {
+        this.status = status;
+    }
+
+    public String getExternalId() {
+        return externalId;
+    }
+
+    public void setExternalId(String externalId) {
+        this.externalId = externalId;
+    }
+
+    public String getInvoiceDate() {
+        return invoiceDate;
+    }
+
+    public void setInvoiceDate(String invoiceDate) {
+        this.invoiceDate = invoiceDate;
+    }
+
+    public String getDueDate() {
+        return dueDate;
+    }
+
+    public void setDueDate(String dueDate) {
+        this.dueDate = dueDate;
+    }
+
+    public String getCreatedAt() {
+        return createdAt;
+    }
+
+    public void setCreatedAt(String createdAt) {
+        this.createdAt = createdAt;
+    }
+
+    public String getUpdatedAt() {
+        return updatedAt;
+    }
+
+    public void setUpdatedAt(String updatedAt) {
+        this.updatedAt = updatedAt;
+    }
+
+    public Amount getTotalAmount() {
+        return totalAmount;
+    }
+
+    public void setTotalAmount(Amount totalAmount) {
+        this.totalAmount = totalAmount;
+    }
+
+    public Amount getPaidAmount() {
+        return paidAmount;
+    }
+
+    public void setPaidAmount(Amount paidAmount) {
+        this.paidAmount = paidAmount;
+    }
+
+    public Amount getDueAmount() {
+        return dueAmount;
+    }
+
+    public void setDueAmount(Amount dueAmount) {
+        this.dueAmount = dueAmount;
+    }
+
+    public List<String> getTags() {
+        return tags;
+    }
+
+    public void setTags(List<String> tags) {
+        this.tags = tags;
+    }
+
+    public List<InvoiceLine> getLines() {
+        return lines;
+    }
+
+    public void setLines(List<InvoiceLine> lines) {
+        this.lines = lines;
+    }
+
+    public String getRecipientId() {
+        return recipientId;
+    }
+
+    public void setRecipientId(String recipientId) {
+        this.recipientId = recipientId;
+    }
+
+    /**
+     * Factory method to generate Invoice object from a String representation of a valid Invoice JSON 
+     * @param invoiceStr
+     * @return
+     * @throws IOException
+     */
+    public static Invoice invoiceFactory(final String invoiceStr) throws IOException {
+        final ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        final JsonNode node = mapper.readTree(invoiceStr);
+        String invoiceId = node.get("invoice").get("id").asText();
+        try{
+            final Invoice invoice = (Invoice)mapper.readValue(node.get("invoice").traverse(), (Class)Invoice.class);
```

Comment:
Why explicit casting is needed here?

Reply:
`mapper.readValue()` returns `Object` type.